src/mds/Server.cc

```diff
@@ -6611,8 +6611,27 @@ void Server::handle_client_setvxattr(const MDRequestRef& mdr, CInode *cur)
     }
     pi.snapnode->last_modified = mdr->get_op_stamp();
     pi.snapnode->change_attr++;
-
     mdr->no_early_reply = true;
+    for (auto child_snaprealm: cur->snaprealm->open_children) {
```

Comment:
I am not totally sure about this, but, would it matter if some directories are auth on another MDS and have snaprealms?

Reply:
It probably does not matter for subvolumes, but just to discuss something generally:

If you have /a/b/c and /a/b/d, and have snap realms for /a, /c and /d and if /d is auth on a different MDS (than /a and /c), that case needs to be handled, yes? Also, if snaprealm exists for /b, then open_children for /a is snaprealm of /b and likewise in a recursive manner. Correct?

Reply:
> It probably does not matter for subvolumes, but just to discuss something generally:
> 
> If you have /a/b/c and /a/b/d, and have snap realms for /a, /c and /d and if /d is auth on a different MDS (than /a and /c), that case needs to be handled, yes? Also, if snaprealm exists for /b, then open_children for /a is snaprealm of /b and likewise in a recursive manner. Correct?

right, that would need to be handled.